setup.py

```diff
@@ -148,6 +148,38 @@ def bundled_settings(debug):
 
     return settings
 
+def check_pkgconfig():
+    """ pull compile / link flags from pkg-config if present. """
+    pcfg = None
+    zmq_config = None
+    try:
+        check_call(['pkg-config', '--exists', 'libzmq'])
+        # this would arguably be better with --variable=libdir /
+        # --variable=includedir, but would require multiple calls
+        pcfg = Popen(['pkg-config', '--libs', '--cflags', 'libzmq'],
+                     stdout=subprocess.PIPE)
+    except OSError as osexception:
+        if osexception.errno == errno.ENOENT:
+            info('pkg-config not found')
+        else:
+            warn("Running pkg-config failed - %s." % osexception)
+    except CalledProcessError:
+        info("Did not find libzmq via pkg-config.")
+
+    if pcfg is not None:
+        output, _ = pcfg.communicate()
```

Comment:
probably want to add `output = output.decode('utf8', 'replace')` on this one, for py3 compat.